web/server/web_client.c

```diff
@@ -1041,7 +1041,7 @@ static inline HTTP_VALIDATION http_request_validate(struct web_client *w) {
                 } else {
                     web_client_split_path_query(w, encoded_url);
 
-                    if (w->separator) {
+                    if (w->separator && w->url_search_path) {
```

Comment:
I don't see how this will fix the corruption, the problem is not `w->url_search_path` being NULL. If it was NULL it would crash immediately. The problem is that it is *not* NULL and it goes and writes someplace where it shouldn't. Maybe it should be initialized to NULL at some other place as well?

Reply:
the web client structure is always set to zero in the file web_client_cache.c in the function present at the line 162, this function is called from static-threaded.c line 16, this is the motive I did not change before, but I am setting it to NULL case I missed something in the whole process.